src/Raven.Server/Rachis/Leader.cs

```diff
@@ -184,15 +189,15 @@ private void RefreshAmbassadors(ClusterTopology clusterTopology, Dictionary<stri
                     if (old.TryGetValue(voter.Key, out FollowerAmbassador existingInstance))
                     {
                         existingInstance.UpdateLeaderWake(_voterResponded);
-                        _voters.Add(voter.Key, existingInstance);
+                        _voters[voter.Key] = existingInstance;
```

Comment:
I think that you wanted to use `.AddOrUpdate`. The `this[key]` is an equivalent of using `.TryAdd`.

Check here: https://github.com/dotnet/corefx/blob/23915a606017197aed1d24960096b532d24d626e/src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs#L888

Reply:
No, this should work as expected.

Reply:
https://msdn.microsoft.com/en-us/library/dd267253(v=vs.110).aspx

Reply:
Correct. I just noticed that they are passing additional flag when this method is used.